File: Segunda Practica Auxiliatura /Agregacion-Composicion/Eje3/Eje3.py
# INCISO A
class Parte:
    def __init__(self, nombre, peso):
        self._nombre = nombre
        self._peso = peso

    def get_nombre(self):
        return self._nombre

    def set_nombre(self, nombre):
        self._nombre = nombre

    def get_peso(self):
        return self._peso

    def set_peso(self, peso):
        self._peso = peso

    def mostrar_info(self):
        print(f"Parte: {self._nombre} - Peso: {self._peso} kg")


class Avion:
    def __init__(self, modelo, fabricante):
        self._modelo = modelo
        self._fabricante = fabricante
        self._partes = []

    def get_modelo(self):
        return self._modelo

    def set_modelo(self, modelo):
        self._modelo = modelo

    def get_fabricante(self):
        return self._fabricante

    def set_fabricante(self, fabricante):
        self._fabricante = fabricante

    def agregar_parte(self, parte):
        self._partes.append(parte)

    def mostrar_avion(self):
        print(f"Modelo del avión: {self._modelo}")
        print(f"Fabricante: {self._fabricante}")
        print("Partes del avión:")
        for parte in self._partes:
            parte.mostrar_info()


# INCISO B
avion = Avion("Boeing 747", "Boeing")
avion.agregar_parte(Parte("Motor", 5000))
avion.agregar_parte(Parte("Ala Izquierda", 3000))
avion.agregar_parte(Parte("Ala Derecha", 3000))
avion.agregar_parte(Parte("Tren de Aterrizaje", 1500))

# INCISO C
avion.mostrar_avion()

File: Segunda Practica Auxiliatura /Agregacion-Composicion/Eje3/Eje3.java
import java.util.ArrayList;
import java.util.List;

// INCISO A
class Parte {
    private String nombre;
    private double peso;

    public Parte(String nombre, double peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public double getPeso() {
        return peso;
    }

    public void setPeso(double peso) {
        this.peso = peso;
    }

    public void mostrarInfo() {
        System.out.println("Parte: " + nombre + " - Peso: " + peso + " kg");
    }
}

class Avion {
    private String modelo;
    private String fabricante;
    private List<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getFabricante() {
        return fabricante;
    }

    public void setFabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public void agregarParte(Parte parte) {
        partes.add(parte);
    }

    public void mostrarAvion() {
        System.out.println("Modelo del avión: " + modelo);
        System.out.println("Fabricante: " + fabricante);
        System.out.println("Partes del avión:");
        for (Parte parte : partes) {
            parte.mostrarInfo();
        }
    }
}

public class Main {
    public static void main(String[] args) {
        // INCISO B
        Avion avion = new Avion("Boeing 747", "Boeing");
        avion.agregarParte(new Parte("Motor", 5000));
        avion.agregarParte(new Parte("Ala Izquierda", 3000));
        avion.agregarParte(new Parte("Ala Derecha", 3000));
        avion.agregarParte(new Parte("Tren de Aterrizaje", 1500));

        // INCISO C
        avion.mostrarAvion();
    }
}
